dipy/tracking/vox2track.pyx

```diff
@@ -162,6 +161,172 @@ def streamline_mapping(streamlines, voxel_size=None, affine=None,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+cdef inline double norm(double x, double y, double z) nogil:
+    cdef double val = sqrt(x*x + y*y + z*z)
+    return val
+
+
+# Changing this to a memview was slower.
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef inline void c_get_closest_edge(double p_x, double p_y, double p_z,
+                                    double d_x, double d_y, double d_z,
+                                    cnp.double_t[:] edge,
+                                    double eps=1.) nogil:
+     edge[0] = floor(p_x + eps) if d_x >= 0.0 else ceil(p_x - eps)
+     edge[1] = floor(p_y + eps) if d_y >= 0.0 else ceil(p_y - eps)
+     edge[2] = floor(p_z + eps) if d_z >= 0.0 else ceil(p_z - eps)
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def streamlines_in_mask(list streamlines,
+                        cnp.uint8_t[:,:,:] mask,
+                        lin, offset):
+    """Filters streamlines based on whether or not they pass through an ROI,
+    using a line-based algorithm for compressed tract.
+
+    Parameters
+    ----------
+    streamlines : sequence
+        A sequence of streamlines.
+    mask : array-like (uint8)
+        A mask used as a target. Non-zero values are considered to be within
+        the target region.
+    lin : array (3, 3)
+        Transpose of the linear part of the mapping to voxel space, (ie
+        ``inv(affine)[:3, :3].T``)
+    offset : array or scaler
+        Offset part of the mapping (ie, ``inv(affine)[:3, 3]``) + ``.5``. The
+        half voxel shift is so that truncating the result of this mapping
+        will give the correct integer voxel coordinate.
+
+    Returns
+    -------
+    in_mask : array of bool
```

Comment:
You should say that you have one bool per streamline, telling if each streamline is going to through the mask. It's not currently clear if it's based on the mask shape or the number of streamlines.